containers/src/Kokkos_DualView.hpp

```diff
@@ -480,12 +500,9 @@ class DualView : public ViewTraits<DataType, Arg1Type, Arg2Type, Arg3Type> {
   ///   the data in either View.  You must manually mark modified data
   ///   as modified, by calling the modify() method with the
   ///   appropriate template parameter.
-  template <class Device>
-  void sync(const typename std::enable_if<
-                (std::is_same<typename traits::data_type,
-                              typename traits::non_const_data_type>::value) ||
-                    (std::is_same<Device, int>::value),
-                int>::type& = 0) {
+  // deliberately passing args by value as they're used multiple times
+  template <class Device, class... Args>
+  void sync_impl(const std::true_type&, Args const&... args) {
```

Comment:
What was the reason to not pass `std::true_type` by value? Also, you are passing `args` by reference which doesn't match the comment above anymore.

Reply:
Good call, let me update those

Reply:
fe6ef47